pymc3_hmm/step_methods.py

```diff
@@ -160,8 +161,10 @@ def astep(self, point, log_lik_fn, inputs):
         # could be missing out on a much more efficient/faster approach to this
         # potentially large computation.
         # state_seqs = np.broadcast_to(np.arange(M, dtype=int)[..., None], (M, N))
-        # log_lik_t = log_lik_fn(state_seqs)
-        log_lik_t = np.stack([log_lik_fn(np.broadcast_to(m, N)) for m in range(M)])
+        if self.log_lik_t is not None:
+            log_lik_t = self.log_lik_t
```

Comment:
This is the right idea, but we still need to recompute the log-likelihood on every call to this function, because the parameters we're sampling will change on each iteration, which&mdash;in turn&mdash;makes `log_lik_fn` produce new values.  

The question is: how do we recompute the `log_lik_fn` steps below in-place (or as close to in-place as possible)?

In other words, `log_lik_fn` is allocating and filling a large length `N` array, but, now, we want it to fill an _existing_ pre-allocated array with the values it computes.

The answer most likely involves creating our own `log_lik_fn` function, instead of using the one provided by PyMC3.  Ultimately, `log_lik_fn` is just a compiled Theano/Aesara function, so we need to in-place updating at that level first; however, that might be pretty straightforward, because I believe we can create a shared variable that `log_lik_fn` updates in-place.

For that matter, we could turn the entire computation (i.e. the `stack` and `range`) into a Theano/Aesara graph, use a shared variable and have it updated in-place, then call that function instead of `log_lik_fn` and pass `ffbs_astep` the shared variable's underlying value.